DOCS-1604: Document Navigation and SLAM methods (#2397)
viamrobotics/docs: docs/mobility/slam/_index.md

{{% /tab %}}
{{< /tabs >}}

### GetProperties

Get information about the current SLAM session.

{{< tabs >}}
{{% tab name="Python" %}}

**Parameters:**

- `timeout` [(Optional\[float\])](https://docs.python.org/library/typing.html#typing.Optional): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.

**Returns:**

> - Tuple[[bool](https://docs.python.org/3/library/functions.html#bool), [MappingMode.ValueType](https://python.viam.dev/autoapi/viam/proto/service/slam/index.html#viam.proto.service.slam.MappingMode)]: A tuple containing two objects: a boolean which indicates whether the session is being run in the cloud, and `MappingMode`, which represents the [various forms of mapping and localizing the current session can perform](/mobility/slam/cartographer/#use-a-live-machine).

Reviewer: Nice!

> - Tuple[[bool](https://docs.python.org/3/library/functions.html#bool), [MappingMode.ValueType](https://python.viam.dev/autoapi/viam/proto/service/slam/index.html#viam.proto.service.slam.MappingMode)]: A tuple containing two objects: a boolean which indicates whether the session is being run in the cloud, and `MappingMode`, which represents the [various forms of mapping and localizing the current session can perform](/mobility/slam/cartographer/#use-a-live-machine).

Reviewer 2: ```suggestion
- Tuple[[bool](https://docs.python.org/3/library/functions.html#bool), [MappingMode.ValueType](https://python.viam.dev/autoapi/viam/proto/service/slam/index.html#viam.proto.service.slam.MappingMode)]: A tuple containing two objects: a boolean which indicates whether the session is being run in the cloud, and `MappingMode`, which represents the [form of mapping and localizing the current session is performing](/mobility/slam/cartographer/#use-a-live-machine).
```

For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/slam/client/index.html#viam.services.slam.client.SLAMClient.get_properties).

```python {class="line-numbers linkable-line-numbers"}
slam_svc = SLAMClient.from_robot(robot=robot, name="my_slam_service")

# Get the properties of your current SLAM session.
slam_properties = await slam_svc.get_properties()
```

{{% /tab %}}
{{% tab name="Go" %}}

**Parameters:**

- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.

**Returns:**

> - [Properties](https://pkg.go.dev/go.viam.com/services/slam#Properties): Information about the current SLAM session.

Reviewer: ```suggestion
- [Properties](https://pkg.go.dev/go.viam.com/rdk/services/slam#Properties): Information about the current SLAM session.
```
Fixes link

  This includes whether the SLAM process is running in the cloud, as well as its mapping mode.
- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.

> For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/services/slam#Properties).

Reviewer: ```suggestion
For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/services/slam#Properties).
```
